### PR TITLE
Implement forms

### PR DESCRIPTION
**Pull request by GordonOtieno**
- Created a method called current_user in your ApplicationController to make current user data available to all controllers.
  to return the current user
- Created forms to perform the following functions:
  - Create a Post on behalf of the current_user (use the method that you created in your ApplicationController).
  - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
  - Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1 +1,51 @@
...
+    <div class="card mb-3 border-primary" style="width: 180px">
+      <img src="<%= @user.photo %>" alt="avatar" />
+    </div>
+    <div class="card-body">
+      <h2><%= @user.name %></h2>
+    </div>
+    <div class="d-flex justify-content-end" style="width: 350px">
+      <p class="card-text">
+        <small class="text-muted"
+          >Number of posts: <%= @user.posts_counter %></small
```
Comment (VICTOR-CHUKWU) [inline app/views/users/show.html.erb]: - [ ] kindly change `posts_counter` to `postsCounter` as it is what you used in your schema

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1 +1,30 @@
...
+  <div style="width: 540px">
+    <% @users.each do |user| %>
+    <div class="card mb-3 border-primary" style="width: 680px">
+      <div class="col-md-3">
+        <img src="<%= user.photo %>" alt="avatar" />
+      </div>
+      <div class="mt-3 d-flex align-items-center flex-column">
+        <h2 class="card-title"><%= link_to user.name , user %></h2>
+        <p class="d-flex justify-content-end text-muted" style="width: 350px">
+          Number of posts: <%= user.posts_counter %>
```
Comment (VICTOR-CHUKWU) [inline app/views/users/index.html.erb]: - [ ]  kindly change `posts_counter` to `postsCounter` as it is what you used in your schema

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,100 @@
...
+        />
+      </div>
+      <div class="col-md-4">
+        <div class="card-body">
+          <h2 class="card-title"><%= @user.name %></h2>
+        </div>
+        <div class="d-flex justify-content-end" style="width: 350px">
+          <p class="card-text">
+            <small class="text-muted"
+              >Number of posts: <%= @user.posts_counter %></small
```
Comment (VICTOR-CHUKWU) [inline app/views/posts/index.html.erb]: - [ ]  kindly change `posts_counter` to `postsCounter` as it is what you used in your schema

File: app/views/posts/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  <div class="mt-3 d-flex align-items-center flex-column">
+    <h1>Add new Post</h1>
+    <%= form_with model: @post, url: user_posts_path, method: :post, class:
+    'form' do |f|%> <%= f.text_field :title, class:"form-control" %>
+    <br />
+    <%= f.text_area :text, class:"form-control" %>
+    <br />
+    <%= f.submit "Post", class:"btn btn-primary" %> <% end %>
+  </div>
+</div>
```
Comment (VICTOR-CHUKWU) [inline app/views/posts/new.html.erb:12]: - [ ] your post is not creating, kindly fix this
Screenshot from 2022-12-08 12-32-57

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @GordonOtieno  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -9,7 +9,7 @@
     <div class="d-flex justify-content-end" style="width: 350px">
       <p class="card-text">
         <small class="text-muted"
-          >Number of posts: <%= @user.posts_counter %></small
+          >Number of posts: <%= @user.postsCounter %></small
         >
       </p>
```
Comment (HAMMAS-SALEEM) [inline app/views/users/show.html.erb:14]: image

**[Optional]**
- [ ] Your link to license file is not working. :fearful: Kindly fix this link to make your readme file descriptive and license file easily accessible for all users :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Hey @GordonOtieno 👋
## Status: Approved :green_circle: :1st_place_medal: :100:
success

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Linters
✔️ Following Requirements

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**